odo/backends/bcolz.py

```diff
@@ -81,7 +81,18 @@ def resource_bcolz(uri, dshape=None, expected_dshape=None, **kwargs):
                              " valid datashape")
         dshape = datashape.dshape(dshape)
 
+        # bcolz don't like unicode (yet). Should we coerce unicode strings
+        # to ascii (bytes)?
+        coerce = kwargs.get('bcolz_coerce_unicode', False)
+        if coerce:
+            if datashape.predicates.isrecord(dshape.measure):
+                for i, ty in enumerate(dshape.parameters[1].types):
+                    ty = getattr(ty, 'ty', ty)
+                    if hasattr(ty, 'encoding'):
+                        ty.encoding = 'A'
```

Comment:
I like the idea here buy I don't think we should be mutating the dshape in place. Can we do a functional update and forward the newly encoded dshape along?